### PR TITLE
feat: add macos universal build ci config

### LINKED ISSUE #113851: Exploration: Universal binary for apple silicon
- [ ] Electron runtime
- [ ] Native modules
- [ ] Anything else ?

* Binary size doubles considering x64 and arm apps are combined.
* There is no impact on startup because there is no translation involved.

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes https://github.com/microsoft/vscode/issues/113851

### CODE REVIEW

**Review by joaomoreno — COMMENTED** [review]
Anyway we can reuse `product-build-darwin.yml`?

**Comment by deepak1556** [comment]
Yup we could reuse it, but it will make it a bit more complex file with added condition based steps. Would that be okay ?

**Comment by joaomoreno** [comment]
That's OK 👍

File: build/package.json [hunk 1]
```diff
@@ -36,13 +36,15 @@
...
+    "fs-extra": "^9.1.0",
     "iconv-lite-umd": "0.6.8",
     "jsonc-parser": "^2.3.0",
     "mime": "^1.4.1",
     "mkdirp": "^1.0.4",
     "source-map": "0.6.1",
     "typescript": "4.2.0-dev.20201207",
-    "vsce": "1.48.0"
+    "vsce": "1.48.0",
+    "vscode-universal": "deepak1556/universal#61454d96223b774c53cda10f72c2098c0ce02d58"
```
Comment (joaomoreno) [inline build/package.json:47]: Can we publish this?

**Review by joaomoreno — COMMENTED** [review]
This was hard to understand from reading in GH. But now I get it: great job!

If this runs fine with all the 4 permutations of the `VSCODE_BUILD_MACOS` and `VSCODE_BUILD_MACOS_ARM64` parameters, then we're good to go.

Sadly, this will increase our build times considerably... so we gotta improve on that soon. cc @lszomoru

**Comment by deepak1556** [comment]
> If this runs fine with all the 4 permutations of the VSCODE_BUILD_MACOS and VSCODE_BUILD_MACOS_ARM64 parameters, then we're good to go.

Yup verified it and also just cleaned up the conditions in the config.

File: build/package.json (hunk 1 above)
Comment (deepak1556) [reply build/package.json:47]: Will follow up on this, there is still some todo left on the package. I want to push the changes the back to upstream with some cleanup and not maintain a fork.

Comment (joaomoreno) [reply build/package.json:47]: Let's make that a P1, since we aim not to have any commit pointers in `package.json`.

Comment (deepak1556) [reply build/package.json:47]: Tracking issue https://github.com/microsoft/vscode/issues/115262

**Pull request merged**